interop-cats/jvm/src/main/scala/scalaz/zio/interop/catszmanagedjvm.scala

```diff
@@ -0,0 +1,102 @@
+package scalaz.zio.interop
+
+import cats.effect.{ Bracket, LiftIO, Resource }
+import cats.{ ~>, effect, Bifunctor, Defer, Functor, Monad, MonadError, Monoid, Semigroup, SemigroupK }
+import scalaz.zio.{ Reservation, ZIO, ZManaged }
+
+trait CatsZManagedSyntax {
+
+  implicit class ResourceSyntax[F[_], A](resource: Resource[F, A]) {
+    def toManaged(
+      implicit nat: F ~> ZIO[Any, Throwable, ?],
+      b: Bracket[F, Throwable],
+      zB: Bracket[ZIO[Any, Throwable, ?], Throwable],
+      d: Defer[ZIO[Any, Throwable, ?]]
+    ): ZManaged[Any, Throwable, A] =
+      ZManaged.unwrap(
+        resource
+          .mapK(nat)
+          .allocated
+          .map {
+            case (acquire, cleanup) =>
+              val reservation = Reservation(ZIO.succeed(acquire), cleanup.orDie) // should we die here or swallow the exception with catchall, etc?
+              ZManaged.reserve(reservation)
+          }
+          .uninterruptible
+      )
+  }
+
+  implicit class ZManagedSyntax[R, A](managed: ZManaged[Any, Throwable, A]) {
```

Comment:
This type parameter `R` is not currently used, but see below note for how you can support it.